x-pack/functionbeat/config/config.go

```diff
@@ -16,7 +16,7 @@ import (
 )
 
 var (
-	functionPattern = "^[A-Za-z][A-Za-z0-9\\-]{0,139}$"
+	functionPattern = "^[A-Za-z][A-Za-z0-9\\-]{0,29}$"
```

Comment:
I have added this change because, as we're appending the function name to the role name, the role name would exceed AWS's [limit](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateRole.html) of 64 characters. I can change the implementation based on the feedback from the maintainers. Other alternative is to add a conditional for creating the role. 

Either ways, the 140 character limit for the function name needs to be reduced as we're already appending the function name in the role name.

Reply:
This is a good point. However, I am bit worried if we change the role name, we might forgot to change this number. Do you mind refactoring it so the length of the pattern depends on the role name prefix? Or at least add a comment to the code about why the max length is 29?

Reply:
I was off by one on the limit, it can be 30 instead of 29. I've updated the regex, tests and added a comment on how I got to the number 30.